### DISCUSSION TITLE
Is there a way to add elements to an intrusive list with only forward_link node types?

### DISCUSSION THREAD

**Discussion by tanvib00** [ETLCPP/etl#791] [Q&A]
The `push_back` and `insert` methods both require the use of `link_type::etl_previous`, so we can't use these methods to add forward_link type nodes to an intrusive list. Is there another way of doing this which doesn't require using only bidirectional_link types? 

Looking into this to save the extra 4 bytes required for the previous pointer saved by bidirectional links.

**Comment by jwellbelove** [comment]
A bidirectional list requires bidirectional nodes.
Why not use an `etl::intrusive_forward_list`?
